File: practice/01-practice/practicum-03-upr1/Lesson3-Person/Completed/Lesson3-exercise2-completed-vb/TimerEvents/TimerEvents/Form1.vb
Public Class Form1
    Dim t As Timer

    Private Sub Form1_Shown(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Shown
        t = New System.Windows.Forms.Timer
        t.Interval = 1000

        AddHandler t.Tick, AddressOf Me.t_Tick
        t.Start()
    End Sub

    Private Sub t_Tick(ByVal sender As Object, ByVal e As EventArgs)
        ProgressBar1.Value += 10
        If ProgressBar1.Value = 100 Then
            t.Stop()
        End If
    End Sub
End Class

File: practice/01-practice/practicum-03-upr1/Lesson3-Person/Completed/Lesson3-exercise2-completed-cs/TimerEvents/TimerEvents/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace TimerEvents
{
    public partial class Timer : Form
    {
        System.Windows.Forms.Timer t;

        public Timer()
        {
            InitializeComponent();
        }

        void t_Tick(object sender, EventArgs e)
        {
            progressBar.Value += 10;

            if (progressBar.Value >= 100)
                t.Stop();
        }

        private void Timer_Shown(object sender, EventArgs e)
        {
            t = new System.Windows.Forms.Timer();
            t.Interval = 1000;
            t.Tick += new EventHandler(t_Tick);
            t.Start();
        }
    }
}